docs/applications/containers/introduction-to-microservices-with-kubernetes-and-gitlab/index.md

```diff
@@ -0,0 +1,210 @@
+
+1. Domain propagation will take a few minutes to successfully resolve to your new IP address following this change, and the `gitlab-runner` microservice may crash since it relies on this resolution. You'll want to periodically check this service to make sure it hasn't failed. See the [Troubleshooting section](/docs/applications/containers/introduction-to-microservices-with-kubernetes-and-gitlab/#troubleshooting) for tips on how to resolve Pods in statuses other than `Running`.
+
+        kubectl get pods | grep gitlab-runner
+
+1. Navigate to the [NodeBalancer](https://cloud.linode.com/nodebalancers) section of the Cloud Manager, and find the IP address of this new resource. The label of the NodeBalancer will be a randomly generated 32 character string, with your worker nodes pre-configured as Backend Linodes.
+
+1. If you have a domain set-up to point to one of your worker nodes as instructed above, as your containers come up you should see A records made automatically for each of the subdomains. They should point to your NodeBalancer. If not, you can create them manually.
+
+  -  minio.example.com
+  -  gitlab.example.com
+  -  registry.example.com
+
+1. Finally, once all services have initialized, you can navigate to `gitlab.example.com` using the domain name that you configured earlier, and after a few moments, you'll see a login screen for your new Microservices Application.
+
+![Gitlab Login](gitlablogin.png)
+
+## Troubleshooting
+
+### Gitlab-runner in CrashLoopbackOff Status
+
+If gitlab-runner is in the `CrashLoopbackOff` status, it has most likely timed out before your domains have resolved, and you'll need to delete the pod once domain propagation is completed for the service to restart and confirm your domain is working. You can do this by replacing the randomly generated string following `gitlab-gitlab-runner-` with the string that appeared in the previous command:
+
+    kubectl delete pods gitlab-gitlab-runner-dd97f84b-zplq9
+
+### Pods Stuck in Initialization Status
+
+Other Pods, such as Sidekiq, Unicorn, or other Rails based containers may be stuck in initialization status. This is usually due to them waiting on the dependencies container.
+
+1. Check the logs of the given Pod. If you see something like the following it's an indication that the migrations haven't completed:
+
+    {{< output >}}
+Checking database connection and schema version
+WARNING: This version of GitLab depends on gitlab-shell 8.7.1, ...
+Database Schema
+Current version: 0
+Codebase version: 20190301182457
+{{</ output >}}
+
+1. Find the migrations job:
+
+        kubectl get job -lapp=migrations
+
+1. Find the Pod run by the Job:
+
+        kubectl get pod -ljob-name=job-name
+
+1. Look at the `STATUS` column. If it's `Completed` then the job should pass checks and the container should start shortly. If it's `Running` check the logs and address any errors:
+
+        kubectl logs pod-name
+
+### UPGRADE FAILED: "$name" has no deployed releases
```

Comment:
I added this section because I needed to do this several times to during my testing and thought it would be helpful.